### PR TITLE
Add K.arange implementation for CNTK backend

### PR DESCRIPTION
**Pull request by SriRangaTarun**
### Summary
I added a CNTK arange function (default type is 'float32' because CNTK constants do not support integer types). I also reopened the arange test for CNTK.
 
### PR Overview

- [n] This PR requires new unit tests [y/n] (make sure tests are included)
- [n] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [y] This PR is backwards compatible [y/n]
- [y] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

**Comment by SriRangaTarun** [comment]
The Travis CI Build throws an error : "PyTest 5.0 will not support Python 2.7" and the python 2.7 test CNTK test fails. Before, this test was being skipped.

**Comment by fchollet** [comment]
Let's restart the CI tests.

**Comment by SriRangaTarun** [comment]
@fchollet The build is now saying this : 
"Replacing crashed worker gw1
[gw2] linux2 Python 2.7.15 cwd: /home/travis/build/keras-team/keras"

**Comment by abhaikollara** [comment]
I was having an issue with creating an empty `arange`. `C.variables.Constant([])` throws a floating point exception. Seems like a problem within CNTK

File: keras/backend/cntk_backend.py [hunk 1]
```diff
@@ -2750,8 +2751,33 @@ def cumprod(x, axis=0):
...
+    # Match the behavior of numpy and Theano by returning an empty sequence.
+    if stop is None:
+        try:
+            if start < 0:
+                start = 0
+        except TypeError:
+            # Handle case where start is a tensor
+            start = 0 if start.value < 0 else start.value
+
+    result = C.variables.Constant(np.arange(start, stop, step),
```
Comment (taehoonlee) [inline keras/backend/cntk_backend.py]: Thank you for the PR, @SriRangaTarun. The following will work:
```python
ctype = _convert_string_dtype(dtype)
return variable(value=np.arange(start, stop, step).astype(ctype), dtype=dtype, name=name)
```

File: keras/backend/cntk_backend.py [hunk 2]
```diff
@@ -13,6 +13,7 @@
 from contextlib import contextmanager
 import warnings
 
+evaluate = eval  # Separate python eval keyword from K.eval
```
Comment (taehoonlee) [inline keras/backend/cntk_backend.py]: Please remove.

Comment (SriRangaTarun) [reply keras/backend/cntk_backend.py]: Fixed

File: keras/backend/cntk_backend.py (hunk 1 above)
Comment (SriRangaTarun) [reply keras/backend/cntk_backend.py]: Fixed

File: keras/backend/cntk_backend.py [hunk 3]
```diff
@@ -2750,8 +2749,32 @@ def cumprod(x, axis=0):
...
+        dtype: Integer dtype to use.
+    # Returns
+        An integer tensor.
+    """
+    # Match the behavior of numpy and Theano by returning an empty sequence.
+    if stop is None:
+        try:
+            if start < 0:
+                start = 0
+        except TypeError:
```
[the remark this answers is not in the document]
Comment (SriRangaTarun) [reply keras/backend/cntk_backend.py]: Fixed

**Comment by SriRangaTarun** [comment]
The build is saying this :
"Replacing crashed worker gw1
[gw2] linux2 Python 2.7.15 cwd: /home/travis/build/keras-team/keras"

**Comment by fchollet** [comment]
Thank you for preparing this PR.

The CNTK backend is no longer in development at this time. We won't be adding making any change going forward.

**Pull request closed without merging**